src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
How is the IPv4 handling of this any different than the IPv6 implementation? With IPv6 all private IPs are allowed, including many trillions+ outside of the user's subnet. Once again, private IPs aren't routable over the internet. Even if a router allowed such a packet through, the nature of a TCP connection wouldn't allow the source of the spoofed IP to receive the reply.